worksheets_test.go

```diff
@@ -84,6 +86,227 @@ func (s *Zuite) TestWorksheetNew_multipleDefsSameName() {
 	}
 }
 
+func (s *Zuite) TestExample_externalComputedBy() {
+	_, err := NewDefinitions(strings.NewReader(`worksheet simple {
+		2:hello_name text computed_by { external }
+	}`))
+	if assert.Error(s.T(), err) {
+		require.Equal(s.T(), "plugins: missing plugin for simple.hello_name", err.Error())
+	}
+}
+
+func (s *Zuite) TestExample_externalComputedBy2() {
+	opt := Options{
+		Plugins: map[string]map[string]ComputedBy{
+			"not_so_simple": map[string]ComputedBy{},
+		},
+	}
+	_, err := NewDefinitions(strings.NewReader(`worksheet simple {
+	}`), opt)
+	if assert.Error(s.T(), err) {
+		require.Equal(s.T(), "plugins: unknown worksheet(not_so_simple)", err.Error())
+	}
+}
+
+func (s *Zuite) TestExample_externalComputedBy3() {
+	opt := Options{
+		Plugins: map[string]map[string]ComputedBy{
+			"simple": map[string]ComputedBy{
+				"unknown_name": nil,
+			},
+		},
+	}
+	_, err := NewDefinitions(strings.NewReader(`worksheet simple {
+	}`), opt)
+	if assert.Error(s.T(), err) {
+		require.Equal(s.T(), "plugins: unknown field simple.unknown_name", err.Error())
+	}
+}
+
+func (s *Zuite) TestExample_externalComputedBy4() {
+	opt := Options{
+		Plugins: map[string]map[string]ComputedBy{
+			"simple": map[string]ComputedBy{
+				"name": nil,
+			},
+		},
+	}
+	_, err := NewDefinitions(strings.NewReader(`worksheet simple {
+		1:name text
+	}`), opt)
+	if assert.Error(s.T(), err) {
+		require.Equal(s.T(), "plugins: field simple.name not externally defined", err.Error())
+	}
+}
+
+func (s *Zuite) TestExample_externalComputedBy6() {
+	opt := Options{
+		Plugins: map[string]map[string]ComputedBy{
+			"simple": map[string]ComputedBy{
+				"name": sayAlice([]string{"age"}),
+			},
+		},
+	}
+	defs, err := NewDefinitions(strings.NewReader(`worksheet simple {
+		1:name text computed_by { external }
+		2:age number[0]
+	}`), opt)
+	require.NoError(s.T(), err)
+
+	ws := defs.MustNewWorksheet("simple")
+
+	err = ws.Set("name", NewText("Alex"))
+	if assert.Error(s.T(), err) {
+		require.Equal(s.T(), "cannot assign to computed field name", err.Error())
+	}
+}
+
+type sayAlice []string
+
+var _ ComputedBy = sayAlice([]string{})
+
+func (sa sayAlice) Args() []string {
+	return sa
+}
+
+func (sa sayAlice) Compute(...Value) Value {
+	return NewText("Alice")
+}
+
+type fullName []string
+
+var _ ComputedBy = fullName([]string{})
+
+func (fn fullName) Args() []string {
+	return fn
+}
+
+func (fn fullName) Compute(values ...Value) Value {
+	firstName := values[0].String()
```

Comment:
Can you add a `// TODO(pascal): we need to figure out how to make Values useful, e.g. having an AsString()`